### ISSUE TITLE
Error when column contains UUID in AWS Glue

### ISSUE THREAD

**Issue by keatmin** [awslabs/aws-data-wrangler#200]
**Describe the bug**

When running wr.catalog.extract_athena_types and wr.db.read_sql_query where there are columns with UUID, it returns an exception. When I circumvent wr.db.read_sql_query with pd.read_sql_query, I run into the the same exception as below: 

Exception with extract_athena_types
```
File "/glue/lib/installation/awswrangler/catalog.py", line 1309, in extract_athena_types
    df=df, index=index, partition_cols=partition_cols, dtype=dtype, index_left=index_left
  File "/glue/lib/installation/awswrangler/_data_types.py", line 292, in athena_types_from_pandas_partitioned
    df=df, index=index, dtype=dtype, index_left=index_left
  File "/glue/lib/installation/awswrangler/_data_types.py", line 270, in athena_types_from_pandas
    df=df, index=index, ignore_cols=list(casts.keys()), index_left=index_left
  File "/glue/lib/installation/awswrangler/_data_types.py", line 248, in pyarrow_types_from_pandas
    for field in pa.Schema.from_pandas(df=df[cols], preserve_index=index):
  File "pyarrow/types.pxi", line 1078, in pyarrow.lib.Schema.from_pandas
  File "/glue/lib/installation/pyarrow/pandas_compat.py", line 521, in dataframe_to_types
    type_ = pa.array(c, from_pandas=True).type
  File "pyarrow/array.pxi", line 265, in pyarrow.lib.array
  File "pyarrow/array.pxi", line 80, in pyarrow.lib._ndarray_to_array
  File "pyarrow/error.pxi", line 84, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: Could not convert 422a0dee-4b98-405d-978c-0e0e96482643 with type UUID: did not recognize Python value type when inferring an Arrow data type
```
**To Reproduce**
Version: 1.0.2 whl and 1.0.4 whl on AWS Glue Python Shell 3
Only happens in AWS Glue unfortunately. Wondering if `object` type should be added into `pyarrow_types_from_pandas` to convert them into `pa.string()`. I can make a PR for that.

**Comment by igorborgest** [comment]
Hi @keatmin, thanks for reaching out!

Unfortunately, we can't force all `object` types to became `string`, because we have others types that are represented the same way on Pandas (e.g. `pa.binary()`, `pa.date32()`, `pa.decimal128()`) .

A couple questions:

- What is your target database? (e.g. PostgreSQL)
- Did you try to pass the dtype argument to cast this column? (e.g. `dtype={'col_name': pa.string()}`)

**Comment by keatmin** [comment]
@igorborgest first of all really appreciate your work in this. 

Yes, that's how I solved it by 
- casting the column in dtype 
-  casting the column to varchar/text in the sql query. 

Target is s3. I still created an issue here because it seems to be only in Glue and want to dive deeper into it to understand. 

Yeah dived into pandas types and realised converting all object to `string` is a really bad idea. 

I can close this.

**Comment by igorborgest** [comment]
Thanks for the quick answer @keatmin.

I will explore how we can improve the experience for this kind of case, could you please provide what is the database you are working on? Is it PostgreSQL?

**Comment by keatmin** [comment]
Yikes sorry, forgot about that. Yes, it is postgresql. 

Also Athena seems to be very fuzzy about datetime and timestamp when uploading in csv. I have to remove timezone from each column to ensure that it's compatible with Athena. Not sure about other db because I have only worked with PostgreSQL, so just curious if there is a need to convert timezone in pandas in other db for it to be Athena friendly

**Comment by igorborgest** [comment]
Reopen to try to address the reported issues in the next minor release!

**Comment by igorborgest** [comment]
@keatmin Thanks again for report it!

We just added support to UUID on PostgreSQL and it will be available int next release `1.1.0`.

*P.S.* Athena does not have full support for timezone on timestamps, so I suggest you to store it in a separated column.

**Comment by keatmin** [comment]
Whoa great work @igorborgest. Thank you so much!

**Comment by igorborgest** [comment]
Available on version [1.1.0](https://github.com/awslabs/aws-data-wrangler/releases/tag/1.1.0)
